File: test-suite/src/test/java/io/micronaut/docs/claimsaudience/CustomClaimsAudienceProvider.java
package io.micronaut.docs.claimsaudience;

import io.micronaut.context.annotation.Requires;
import io.micronaut.security.token.claims.ClaimsAudienceProvider;
import jakarta.inject.Singleton;

import java.util.List;

@Requires(property = "spec.name", value = "claims-generation-docs")
//tag::clazz[]
@Singleton
class CustomClaimsAudienceProvider implements ClaimsAudienceProvider {
    private static final List<String> AUDIENCE = List.of("https://api.example.com");

    @Override
    public List<String> audience() {
        return AUDIENCE;
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/claimsaudience/CustomClaimsAudienceProvider.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.security.token.claims import ClaimsAudienceProvider

AUDIENCE = ["https://api.example.com"]


@Requires(property="spec.name", value="claims-generation-docs")
# tag::clazz[]
@Singleton
class CustomClaimsAudienceProvider(ClaimsAudienceProvider):

    def audience(self) -> list[str]:
        return AUDIENCE
# end::clazz[]
